res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,426 @@
+        engine.setValue("[EffectRack1_EffectUnit2]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][1]);
+        Mixage.toggleLED(isEnabled ? 1 : 0, "[Channel" + unitNr + "]", "fx_on");
+    }
+};
+
+// The FX SEL button that selects which effects are enabled for channel 1/2
+Mixage.handleEffectChannelSelect = function(channel, control, value, _status, _group) {
+    // calculate effect unit number from MIDI control. 0x07 controls unit 1, 0x15 unit 2
+    var unitNr = control === 0x07 ? 1 : 2;
+    // react only on first message / keydown
+    if (value === 0x7F) {
+        // check and toggle select state
+        var selected1 = Mixage.effectRackSelected[unitNr][0];
+        var selected2 = Mixage.effectRackSelected[unitNr][1];
+        if (selected1 && selected2) {
+            selected1 = true;
+            selected2 = false;
+        } else if (selected1) {
+            selected1 = false;
+            selected2 = true;
+        } else {
+            selected1 = true;
+            selected2 = true;
+        }
+        Mixage.effectRackSelected[unitNr][0] = selected1;
+        Mixage.effectRackSelected[unitNr][1] = selected2;
+        // update controls
+        var isEnabled = Mixage.effectRackEnabled[unitNr];
+        var keyString = "group_[Channel" + unitNr + "]_enable";
+        engine.setValue("[EffectRack1_EffectUnit1]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][0]);
+        engine.setValue("[EffectRack1_EffectUnit2]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][1]);
+    }
+};
+
+Mixage.handleEffectDryWet = function(channel, control, value, _status, _group) {
+    // calculate effect unit number from MIDI control. 0x21 controls unit 1, 0x25 unit 2
+    var unitNr = control === 0x21 ? 1 : 2;
+    // Control centers on 0x40 (64), calculate difference to that value and multiply by 4
+    var diff = (value - 64) / 10.0;
+    // In either case, register the movement
+    var controlString = "[EffectRack1_EffectUnit" + unitNr + "]";
+    var dryWetValue = engine.getValue(controlString, "mix");
+    engine.setValue(controlString, "mix", dryWetValue + diff);
+};
+
+// The PAN rotary control used here for the overall effect super control
+Mixage.handleEffectSuper = function(channel, control, value, _status, _group) {
+    // calculate effect unit number from MIDI control. 0x60 controls unit 1, 0x62 unit 2
+    var unitNr = control === 0x60 ? 1 : 2;
+    // Control centers on 0x40 (64), calculate difference to that value and multiply by 4
+    var diff = (value - 64) / 10.0;
+    // In either case, register the movement
+    var controlString = "[EffectRack1_EffectUnit" + unitNr + "]";
+    var mixValue = engine.getValue(controlString, "super1");
+    engine.setValue(controlString, "super1", mixValue + diff);
+};
+
+// The BEAT MOVE rotary control is used as an extra "shift" key when pushed down
```

Comment:
what about the shift button on the hardware? How is that being handled?

Reply:
Controls send different control values when shift is pressed. This is accounted for.